### PR TITLE
Add SIRIUS ms software to conda-forge

### PR DESCRIPTION
**Pull request by mfleisch**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mfleisch** [comment]
@conda-forge/core The test of the Windows build hangs at the package installation step without throwing any error. 
 
 ```
 TEST START: C:\Miniconda\conda-bld\win-64\sirius-ms-4.9.4-h57928b3_0.tar.bz2
Renaming work directory 'C:\Miniconda\conda-bld\sirius-ms_1632395702883\work' to 'C:\Miniconda\conda-bld\sirius-ms_1632395702883\work_moved_sirius-ms-4.9.4-h57928b3_0_win-64'
shutil.move(work)=C:\Miniconda\conda-bld\sirius-ms_1632395702883\work, dest=C:\Miniconda\conda-bld\sirius-ms_1632395702883\work_moved_sirius-ms-4.9.4-h57928b3_0_win-64)
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done

## Package Plan ##

  environment location: C:\Miniconda\conda-bld\sirius-ms_1632395702883\_test_env

The following NEW packages will be INSTALLED:

    sirius-ms: 4.9.4-h57928b3_0 local

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
 ```
 
 Does anybody has a suggestion how to further investigate this issue?

**Comment by mfleisch** [comment]
@conda-forge-admin, please ping team

File: conda-forge.yml
Comment (chrisburr) [inline conda-forge.yml:2]: Please don't make any changes outside of the `recipes/$PKG_NAME` folder when making PRs to staged-recipes.

File: recipes/sirius-ms/meta.yaml
Comment (chrisburr) [inline recipes/sirius-ms/meta.yaml:19]: I'm not very familiar with java but it seems like the JRE is being included in this package? Shouldn't this be depending on `openjdk` instead?

**Comment by bernt-matthias** [comment]
Hey @mfleisch .. do you want to use this opportunity to finish this recipe. There are two open questions as far as I see: 

- the inclusion of JRE https://github.com/conda-forge/staged-recipes/pull/16235#discussion_r830435773
- if the patch I added to bioconda should be included here as well?

**Comment by mfleisch** [comment]
Hey @bernt-matthias,
Thanks for bringing this up again. I think there is not much to do from our side here. The PR is ready to be merged. I am just waiting for it to be reviewed. @chrisburr @carterbox, please let me know if there is something missing from my side that prevents a final review.  

> * if the patch I added to bioconda should be included here as well?     

The tests run without the patch you created for bioconda.

As soon as this is merged and the feedstock isavailable I will start working an updated package for SIRIUS 5. 
This SIRIUS 5 package can then be provided with dependency to the [openjdk-17](https://github.com/bgruening/openjdk-feedstock/tree/jdk17).

**Comment by bernt-matthias** [comment]
> The tests run without the patch you created for bioconda.

True. OK lets see :)

> As soon as this is merged and the feedstock isavailable I will start working an updated package for SIRIUS 5. This SIRIUS 5 package can then be provided with dependency to the [openjdk-17](https://github.com/bgruening/openjdk-feedstock/tree/jdk17).

Sounds great.

**Comment by eeko-kon** [comment]
@chrisburr It would help a lot if you could accept this PR :)

**Comment by bernt-matthias** [comment]
@mfleisch Since there seems to be no interest in having this in conda-forge (I also wrote 2 more messages on the gitter channel) I would suggest to keep this in bioconda? 

Since this is a noarch package it should work on windows anyway. Also bioconda has the advantage that you get a  container for free.

**Comment by bernt-matthias** [comment]
@mfleisch and @eeko-kon please ping @bgruening if you still want this to be merged

**Comment by carterbox** [comment]
Please ping @conda-forge/help-java whenever you want a fresh review. Otherwise, your PR remains labeled "Awaiting author"

**Review by carterbox — CHANGES_REQUESTED** [review]
Please address all previous review comments before asking for a new review.

In addition, this recipe is revending pre-compiled libraries. Conda-forge requires that all binaries are compiled with our infrastructure. Please download the `sirius` source files and compile all libraries/link to exisiting conda-forge packages.

File: recipes/sirius-ms/meta.yaml
Comment (mfleisch) [reply recipes/sirius-ms/meta.yaml:19]: The package does no longer contains a bundled jre but uses  https://github.com/conda-forge/openjdk-feedstock instead.

**Review by mfleisch — COMMENTED** [review]
1. JDK is no longer included. https://github.com/conda-forge/openjdk-feedstock is used instead.
2.  coin-or-clp native libs are  no longer included.  https://github.com/conda-forge/coin-or-cbc-feedstock is used instead. 
3. Java application is build from source using conda-forge build servers and no longer modifies a pre-build package.

File: conda-forge.yml
Comment (mfleisch) [reply conda-forge.yml]: DONE

**Comment by mfleisch** [comment]
@conda-forge/help-java

File: recipes/sirius-ms/meta.yaml
Comment (carterbox) [inline recipes/sirius-ms/meta.yaml:35]: Because you are redistributing many of the java libraries that your package depends on, you need to include their licenses in this section. I think there is a gradle-plugin/tool that can collect the licenses automatically into a single file, but I don't know what it's called because I do not use java.

i.e. license information for amqp, annotations, api-common, base64, batik, bcprov, beam, blob-storage, canopus, ...

File: recipes/sirius-ms/meta.yaml
Comment (carterbox) [inline recipes/sirius-ms/meta.yaml:7]:
```suggestion
{% set version = "5.6.4" %}
{% set commit = "cbc32a4754e0595baed0575f86c95bb4198098f4" %}

package:
  name: sirius-ms
  version: {{ version }}
```

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/sirius-ms/meta.yaml
Comment (mfleisch) [reply recipes/sirius-ms/meta.yaml:17]: We do not ship any native libs, so I just disabled detections where I could to hopefully speed up the build.
I can also remove them. Seems not to make a difference regarding build time anyways. @carterbox Please let me know what you prefer.

File: recipes/sirius-ms/meta.yaml
Comment (mfleisch) [reply recipes/sirius-ms/meta.yaml]: The gradle build now creates a `licenses.md` file that lists the license information of each package included in SIRIUS.

File: recipes/sirius-ms/meta.yaml
Comment (mfleisch) [reply recipes/sirius-ms/meta.yaml]: For now I removed the parameters.

**Comment by mfleisch** [comment]
@conda-forge/help-java

File: recipes/sirius-ms/meta.yaml
Comment (carterbox) [inline recipes/sirius-ms/meta.yaml:28]: Last thing to do is to update this license line with all the licenses that you have collected. You can use OTHER if the license does not have an identifier.

```suggestion
  license: AGPL-3.0-only AND MIT AND ....
```
Docs page: https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions
Identifiers: https://spdx.org/licenses/

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/sirius-ms/meta.yaml
Comment (mfleisch) [reply recipes/sirius-ms/meta.yaml:28]: Many of them have an identifier but not all of them. So I would use the  `AND OTHER`  as you suggested.

**Comment by mfleisch** [comment]
@conda-forge/help-java

**Pull request merged**
